### PR TITLE
Trus 475 - What is the Agency's telephone number?'

### PR DESCRIPTION
**Pull request by JoshuaH01**
Created agencys telephone number page, added AffinityGroupChecker.

### CODE REVIEW

File: app/controllers/AgencysTelehponeNumberController.scala [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+        value => {
+          for {
+            updatedAnswers <- Future.fromTry(request.userAnswers.set(AgencysTelehponeNumberPage, value))
+            _              <- sessionRepository.set(updatedAnswers)
+          } yield Redirect(navigator.nextPage(AgencysTelehponeNumberPage, mode)(updatedAnswers))
+        }
+      )
+  }
+}
```
Comment (jassalrichy) [inline app/controllers/AgencysTelehponeNumberController.scala]: rename "Agencys" to "Agent" throughout

File: test/views/AgencysTelehponeNumberViewSpec.scala [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    def applyView(form: Form[_]): HtmlFormat.Appendable =
+      view.apply(form, NormalMode)(fakeRequest, messages)
+
+    behave like normalPage(applyView(form), messageKeyPrefix)
+
+    behave like pageWithBackLink(applyView(form))
+
+    behave like stringPage(form, applyView, messageKeyPrefix, routes.AgencysTelehponeNumberController.onSubmit(NormalMode).url)
+  }
+}
```
Comment (jassalrichy) [inline test/views/AgencysTelehponeNumberViewSpec.scala]: Pass "hint" param within normalPage behavious

Add behaviour pageWithASumbitButton

File: test/forms/AgencysTelehponeNumberFormProviderSpec.scala [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      requiredError = FormError(fieldName, requiredKey)
+    )
+
+    behave like nonEmptyField(
+      form,
+      fieldName,
+      requiredError = FormError(fieldName, requiredKey, Seq(fieldName))
+    )
+  }
+}
```
Comment (jassalrichy) [inline test/forms/AgencysTelehponeNumberFormProviderSpec.scala]: Add behave like fieldWithRegexpWithGenerator

File: test/views/AgencysTelehponeNumberViewSpec.scala (hunk 2 above)
Comment (JoshuaH01) [reply test/views/AgencysTelehponeNumberViewSpec.scala]: Done

File: test/forms/AgencysTelehponeNumberFormProviderSpec.scala (hunk 3 above)
Comment (JoshuaH01) [reply test/forms/AgencysTelehponeNumberFormProviderSpec.scala]: done

File: app/controllers/AgencysTelehponeNumberController.scala (hunk 1 above)
Comment (JoshuaH01) [reply app/controllers/AgencysTelehponeNumberController.scala]: done

File: test/views/AgencysTelehponeNumberViewSpec.scala (hunk 2 above)
Comment (JoshuaH01) [reply test/views/AgencysTelehponeNumberViewSpec.scala]: Done

File: test/forms/AgencysTelehponeNumberFormProviderSpec.scala (hunk 3 above)
Comment (JoshuaH01) [reply test/forms/AgencysTelehponeNumberFormProviderSpec.scala]: done

**Pull request merged**